README.md

```diff
@@ -1,61 +1,97 @@
+
+### Architecture
+
+DirectPV is designed to be lightweight and scalable to 10s of 1000s of drives. It is made up of three components:
+
+- Controller
+- Node Driver
+- UI
+
+![DirectPV Architecture](./directpv_architecture.png)
+
+##### Controller
 
-Historically, these datastores relied on Kubernetes-provided HostPath and LocalPV for using locally attached drives. Local and Host Path Volume drivers are manual, static and ephemeral, introducing management complexity. 
+When a volume claim is made, the controller provisions volumes uniformly from a pool free drives. DirectPV is aware of pod's affinity constraints, and allocates volumes from drives local to pods. 
 
-### QuickStart
+_Only one active instance of controller runs per cluster._
 
-Here is the ***extremely*** quickstart:
+##### Node Driver
+
+Node Driver implements the volume management functions such as discovery, format, mount, and monitoring of drives on the nodes. One instance of node driver runs on each of the storage servers. 
+
+_One instance of node driver runs on every node._
+
+##### UI
+
+Storage Administrators can use the kubectl CLI plugin to select, manage and monitor drives. Web based UI is currently under development. 
+
+### Installation
 
 ```sh
+# Install kubectl directpv plugin
 kubectl krew install directpv
+
+# Use the plugin to install directpv in your kubernetes cluster
 kubectl directpv install
+
+# Ensure directpv has successfully started
+kubectl directpv info
+
+# List available drives in your cluster
 kubectl directpv drives ls
-# choose all the drives that directpv should manage and format them
+
+# Select drives that directpv should manage and format
 kubectl directpv drives format --drives $DRIVE_SELECTOR_ELLIPSES --nodes $NODE_SELECTOR_ELLIPSES
-# 'directpv-min-io' can now be specified as the storageclass in PodSpec.VolumeClaimTemplates
+
+# 'directpv' can now be specified as the storageclass in PodSpec.VolumeClaimTemplates
 ```
 
-For more information, please visit our [documentation](./docs/index.md).
+For air-gapped setups and advanced installations, please refer to the [Installation guide](./docs/installation.md).
 
-### How is it different from LocalVolume provisioner?
+### Upgrade
 
-[LocalVolume](https://kubernetes.io/blog/2019/04/04/kubernetes-1.14-local-persistent-volumes-ga/) provisioner also allows direct access to the storage medium. However, it requires manual management of drives and volumes - i.e. it does not support dynamic provisioning of volumes, storage management or dynamic scheduling of pods on nodes with available volumes. 
+DirectPV version upgrades are seameless and transparent. Simply uninstall an existing version of directpv and install with a newer version to upgrade.
```

Comment:
```suggestion
DirectPV version upgrades are seameless and transparent. Simply uninstall an existing version of directpv and install with a newer version to upgrade. For DirectCSI installations, DirectPV provides a binary replacement for DirectCSI starting with 2.0.0. 
```

Reply:
The upgrade procedure does not change when going from direct-csi to directpv